src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -186,12 +188,35 @@ public IEnumerable<IDiagnostic> GetAllDiagnostics()
             return AssembleDiagnostics();
         }
 
-        private IEnumerable<IDiagnostic> AssembleDiagnostics()
+        private IReadOnlyList<IDiagnostic> AssembleDiagnostics()
         {
-            return GetParseDiagnostics()
-            .Concat(GetSemanticDiagnostics())
-            .Concat(GetAnalyzerDiagnostics())
-            .OrderBy(diag => diag.Span.Position);
+            var diagnostics = GetParseDiagnostics()
+                .Concat(GetSemanticDiagnostics())
+                .Concat(GetAnalyzerDiagnostics())
+                .OrderBy(diag => diag.Span.Position);
+            var filteredDiagnostics = new List<IDiagnostic>();
+
+            var disabledDiagnosticsCache = SourceFile.DisabledDiagnosticsCache;
+            foreach (IDiagnostic diagnostic in diagnostics)
+            {
+                (int diagnosticLine, _) = TextCoordinateConverter.GetPosition(SourceFile.LineStarts, diagnostic.Span.Position);
+
+                if (diagnosticLine == 0)
+                {
+                    filteredDiagnostics.Add(diagnostic);
+                    continue;
+                }
+
+                if (disabledDiagnosticsCache.TryGetDisabledNextLineDirective(diagnosticLine - 1) is DisableNextLineDirectiveEndPositionAndCodes disableNextLineDirectiveEndPositionAndCodes &&
+                    disableNextLineDirectiveEndPositionAndCodes.diagnosticCodes.Contains(diagnostic.Code))
```

Comment:
[nit] FYI you can save some space by changing:
```
disabledDiagnosticsCache.TryGetDisabledNextLineDirective(diagnosticLine - 1) is DisableNextLineDirectiveEndPositionAndCodes disableNextLineDirectiveEndPositionAndCodes
```
to:
```
disabledDiagnosticsCache.TryGetDisabledNextLineDirective(diagnosticLine - 1) is { } disableNextLineDirectiveEndPositionAndCodes
```

See https://github.com/dotnet/roslyn/blob/main/docs/features/patterns.md#patterns for info on the syntax

Reply:
That's super cool. Updated here- https://github.com/Azure/bicep/commit/e7fbc952c497ad91fc2d2e8b2ce651756845d9cc. Thanks!